### PR TITLE
[KYUUBI #476] Add ignore list and and restrict list in server

### PR DESCRIPTION
**Pull request by jiwq**
jiwq [Closes #505](https://github.com/yaooqinn/kyuubi/pull/505)    Test Plan Feature [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_
Add ignore list and and restrict list in server

### _How was this patch tested?_
Run local test

### CODE REVIEW

**Comment by turboFei** [comment]
Could you execute this cmd locally to update doc?
```
build/mvn test -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration
```

BTW, GA issue has been fixed, could you rebase the code base?

**Comment by jiwq** [comment]
> Could you execute this cmd locally to update doc?
> 
> ```
> build/mvn test -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration
> ```
> 
> BTW, GA issue has been fixed, could you rebase the code base?

Done. BTW, before execute the above cmd we should `export KYUUBI_UPDATE=1`.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -427,6 +427,20 @@ object KyuubiConf {
...
+      .stringConf
+      .createOptional
+
+  val SESSION_CONF_RESTRICT_LIST: OptionalConfigEntry[String] =
+    buildConf("session.conf.restrict.list")
+      .doc("Restrict the session if the user's session conf has the key.")
+      .version("1.2.0")
+      .stringConf
+      .createOptional
+
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:443]: I think those 2 keys should be `ConfigEntry[Seq[String]]`?

> "Restrict the session if the user's session conf has the key."

And the `restrict` in description a little bit confuse me, I think `fail` may be more easy to understand?

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**